packages/core/bin/generate-ui-types

```diff
@@ -29,13 +29,85 @@ function findReferencedMethods(obj, refs = {}, latestName = '') {
 	return refs;
 }
 
+function generateHttpRequestVariantNodes(knownCredentials, credentialTypes, httpNode, version) {
+	return Object.values(credentialTypes)
+		.filter((credential) => 'httpRequestNodeVariant' in credential.type)
+		.map((credential) => {
+			const credentialType = credential.type;
+			const hasSupportedNodes = knownCredentials[credential.name]?.supportedNodes?.length > 0;
+			const httpIcon = httpNode.description.iconUrl;
+			const httpNodeClone = new httpNode.constructor().getNodeType(version);
```

Comment:
do we really need to clone the entire node properties? can we specify it inherits a specific node and version and update the n8n to support that instead of returning 20 different versions of the same node? 
as this could potentially explode in number in the future.. for example if we release a new version of http node and want to use that here, the number of imported nodes would double

Reply:
@netroy would love your feedback on this.. since you have given a lot of thought to n8n memory problems..

Reply:
This does indeed add a number of new nodes to `types/nodes.json` (though I also fixed a bug to not include duplicate nodes 😄). I'm currently trying to see if I can add some kind of `extend` feature to Nodes like we have for credentials.

Reply:
This does however does change the content of these files, so I still need to validate that those changes don't break any existing functionality.

Reply:
@netroy would not this also be pulling these nodes during execution?

Reply:
@elsmr if @netroy does not have a problem with this here, feel free to ignore my feedback.. hopefully you have not wasted a lot of time on this..

Reply:
just FYI: I still haven't gotten around to reviewing this yet, my earlier comment is only about the memory-usage concerns.

Reply:
Just in time, I was about to have another look at it today, but I'll hold off until you review the changes to `packages/nodes-base/dist/types` @netroy